### ISSUE TITLE
issues with exe mode and new relative imports

### ISSUE THREAD

**Issue by rcoreilly** [go-python/gopy#257]
@justinfx moving the discussion here to this issue ticket instead of the PR..

I'm using `gopy exe` mode -- it builds everything into a subdir, then I do a standard python install which puts everything into the site-packages, which it seems to know to look in without setting anything in PYTHONPATH -- see error message below -- it is finding `/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/gi/go.py`

The -name of the package is `gi` for https://github.com/goki/gi -- building in the python directory there.

I go to run the `examples/widgets/widget.py` file, using the `pygi` executable that was built from exe mode, which has this import:

```
from gi import go, gi, gist, giv, units, ki, mat32, gimain
```

and it gives me an error on the new go.py relative import:

```
oreilly@shadow:~/goki/gi/examples/widgets/ > ./widgets.py 
Traceback (most recent call last):
  File "/Users/oreilly/go/src/github.com/goki/gi/examples/widgets/./widgets.py", line 16, in <module>
    from gi import go, gi, gist, giv, units, ki, mat32, gimain
  File "/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/gi/go.py", line 12, in <module>
    from . import _gi
ImportError: cannot import name '_gi' from 'gi' (/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/gi/__init__.py)
```

Here's the relevant part of that `go.py` file:

```
import collections
try:
	import collections.abc as _collections_abc
except ImportError:
	_collections_abc = collections
from . import _gi
```
per your recent changes

previously, it used to have:

```
import _gi, collections
```

which worked

So basically it is not finding the _gi library, which in the case of exe is built directly into the pygi executable!

I'm guessing we just need to special-case this for the exe version, so it does the `import _gi` instead of the new `from . import _gi`?

Also, some other files are getting the double-dot relative import:

```
from ..gi import _gi
from ..gi import go
```

**Comment by justinfx** [comment]
> I'm using gopy exe mode 

Hmm, yea I never tested exe mode and still wasn't entirely clear what it was used for...

>  I do a standard python install which puts everything into the site-packages, which it seems to know to look in without setting anything in PYTHONPATH

That makes sense, because in python3 everything is an absolute import from the PYTHONPATH path, or using dot-imports within a package that is in the PYTHONPATH.

> `from . import _gi`
> So basically it is not finding the _gi library, which in the case of exe is built directly into the pygi executable!

Technically that would work assuming the compiled python extension module is in the same location as the python source. But in the case of this exe mode, I actually have no idea how this ìmport _gi` actually works. Without looking into the source I am assuming that the executable is build and `_gi` is directly imported into the root namespace, which makes the absolute import just succeed. Maybe this is a good time to explain what the purpose of exe mode actually is and whether that original issue is no longer relevant with all the recent improvements?

> I'm guessing we just need to special-case this for the exe version, so it does the import _gi instead of the new from . import _gi?

I did happen to leave an escape-hatch for the old relative import py2 approach, if you set `-package-prefix` flag to an empty string:
https://github.com/go-python/gopy/commit/8ab2cac1319cf607a7ca6b0831b738c7c3a09d50#diff-e10869882cb41362074fc2654908776e550c81ba2d0ddaae96a7b1ad0aec3a19R669

> Also, some other files are getting the double-dot relative import:

This could be a bug caused by this default case, which I think must be doing some incorrect string formatting based on the value of the name and package prefix values:
https://github.com/go-python/gopy/commit/8ab2cac1319cf607a7ca6b0831b738c7c3a09d50#diff-e10869882cb41362074fc2654908776e550c81ba2d0ddaae96a7b1ad0aec3a19R685

**Comment by rcoreilly** [comment]
exe mode is unfortunately needed in my use-case because the GUI must run in the first thread, and I couldn't get python to give up that first thread and run in some other thread..   So in exe mode I can claim the first thread and then launch the python interpreter after that.

I tried `-package-path=""` and it did fix the import for go.py but all the others have the problem with the double-dot thing:

```
from .gi import _gi
from .gi import go
```

If you have a ready fix for that, so it also does the basic import for package-path="", and fixes the double-dots (assuming those are bad), that would be great!  If not, I can try to look into it.

**Comment by justinfx** [comment]
That's good to hear that the prefix flag fixed that import. I'm not sure if that means that it should be made a default case for exe mode? If the exe thing handles importing a bundled extension then it probably doesn't need package path prefixing. That feature was really more for the standard package importing cases.

I don't have a ready fix but it does look like it is poor string formatting on my part in that previously linked commit. Double dots normally mean two levels up in the package path instead of one. But here it is just incorrect formatting when the name value is empty. I think it shouldn't even add it to an empty string.

**Comment by rcoreilly** [comment]
btw also fixed default case to not add the "." too.
